src/components/NewCommentForm.tsx

```diff
@@ -2,36 +1,135 @@
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import classNames from 'classnames';
+import * as commentService from '../api/comments';
+import { Comment } from '../types/Comment';
+import { Post } from '../types/Post';
+
+type Props = {
+  setComments: Dispatch<SetStateAction<Comment[]>>;
+  setHasError: (err: boolean) => void;
+  post: Post;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  setComments,
+  setHasError,
+  post,
+}) => {
+  //#region states
+  const [name, setName] = useState('');
+  const [hasNameError, setHasNameError] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [hasEmailError, setHasEmailError] = useState(false);
+
+  const [body, setBody] = useState('');
+  const [hasBodyError, setHasBodyError] = useState(false);
+
+  const [isLoading, setIsLoading] = useState(false);
+  //#endregion
+
+  //#region handlers
+  function handleNameChange(ev: React.ChangeEvent<HTMLInputElement>) {
+    setName(ev.target.value);
+    setHasNameError(false);
+  }
+
+  function handleEmailChange(ev: React.ChangeEvent<HTMLInputElement>) {
+    setEmail(ev.target.value);
+    setHasEmailError(false);
+  }
+
+  function handleBodyChange(ev: React.ChangeEvent<HTMLTextAreaElement>) {
+    setBody(ev.target.value);
+    setHasBodyError(false);
+  }
+  //#endregion
+
+  //#region submit & reset
+  function resetOnSubmit() {
+    setBody('');
+
+    setHasNameError(false);
+    setHasEmailError(false);
+    setHasBodyError(false);
+  }
+
+  function resetOnClear() {
+    setName('');
+    setEmail('');
+    setBody('');
+
+    setHasNameError(false);
+    setHasEmailError(false);
+    setHasBodyError(false);
+  }
+
+  function handleSubmit(ev: React.FormEvent) {
+    ev.preventDefault();
+
+    setHasNameError(!name);
+    setHasEmailError(!email);
+    setHasBodyError(!body);
```

Comment:
Consider adding more detailed validation for the email format and possibly for the name and body fields to ensure data integrity before submission.